src/controllers/auth.js

```diff
@@ -93,4 +93,30 @@ module.exports = {
       res.status(422).json({ message: err.message ?? err });
     }
   },
+  googleCallback: async (req, res) => {
+    const cookieAge = 14 * 24 * 3600 * 1000; // 14 days in milliseconds
+    const { _id, username, email, firstName, lastName } = req.user;
+
+    const payload = {
+      user: {
+        username,
+        email,
+        firstName,
+        lastName,
+        _id,
+      },
+    };
+
+    const token = jwt.sign(payload, process.env.JWT_SECRET, {
+      expiresIn: cookieAge,
+      subject: _id.toString(),
+    });
+
+    res.cookie('token', token, {
+      maxAge: cookieAge,
+      httpOnly: true,
+    });
+
+    res.redirect('/api/users');
```

Comment:
I was going to redirect the user to the home page, but it does not exist for the moment. Initially I thought of sending the user object in the response but since there is a redirection to the google page the front end can't send a normal fetch request and expect a data in the response, they need to redirect him to `/api/auth/google`, and since the user left the front end home page, how will the front end be able to return the user to the home page if I don't redirect him from here?

Reply:
Great point, the way this is usually done is:
1. You can save the frontend redirect URL in your env variables and use that.
2. Or you can save the frontend URL from the request headers before starting the google login process.

Here's a [StackOverflow example](https://stackoverflow.com/questions/13335881/redirecting-to-previous-page-after-authentication-in-node-js-using-passport-js).